pkg/nfd-master/nfd-api-controller.go

```diff
@@ -66,6 +72,14 @@ func newNfdController(config *restclient.Config, nfdApiControllerOptions nfdApiC
 		updateNodeFeatureGroupChan:     make(chan string),
 	}
 
+	labelMap, err := metav1.LabelSelectorAsSelector(nfdApiControllerOptions.NodeFeatureNamespaceSelector)
+	if err != nil {
+		klog.ErrorS(err, "failed to convert label selector to map", "selector", nfdApiControllerOptions.NodeFeatureNamespaceSelector)
+		return nil, err
+	}
+
+	c.namespaceLister = NewNamespaceLister(nfdApiControllerOptions.K8sClient, labelMap)
```

Comment:
One possible optimization would be that only start the namespace lister if `opts.NodeFeatureNamespaceSelector != nil`. Then short circuit `isNamespaceSelected()` to return true if `c.namespaceLister == nil`, and skip stop() in `c.stop()`. Doesn't need to be in this PR but you could possibly add a TODO comment. WDYT?

Reply:
makes sense, thanks for the suggestion